packages/next/client/index.js

```diff
@@ -323,14 +329,70 @@ function renderReactElement(reactEl, domEl) {
 
   if (onPerfEntry && ST) {
     try {
-      const observer = new PerformanceObserver(list => {
-        list.getEntries().forEach(onPerfEntry)
-      })
-      // Start observing paint entry types.
-      observer.observe({
-        type: 'paint',
-        buffered: true,
-      })
+      function getSupportedTypes(types) {
+        if (PerformanceObserver && PerformanceObserver.supportedEntryTypes) {
+          return types.filter(type =>
+            PerformanceObserver.supportedEntryTypes.includes(type)
+          )
+        } else {
+          return []
+        }
+      }
+
+      function observeSupportedTypes(entryTypes) {
+        const supportedTypes = getSupportedTypes(entryTypes)
+
+        /**
+         * Layout shift has special handling as this will be sent
+         * cumulative layout shift.
+         */
+        if (supportedTypes.includes('layout-shift')) {
```

Comment:
Nit: it feels strange to me to put the `layout-shift` exception inside a function called `observeSupportedTypes`. CLS is the metric, and `layout-shift` is the entry, so if the function's job is to observe entries, I'd make a separate function to calculate the metric from that.

Reply:
done